File: Top_Interview_150_List_With_solution(leetcode)/Array|String/L13-roman_to_integer.py
# https://leetcode.com/problems/roman-to-integer/
class Solution:
    def romanToInt(self, s: str) -> int:
        romanLetters = {"I" : 1, "V" : 5, "X" : 10, "L" : 50, "C" : 100, "D" : 500, "M" : 1000, "IV" : 4, "IX" : 9, "XL" : 40, "XC" : 90, "CD" : 400, "CM" : 900}
        val = 0
        i = 0
        c=""
        while i < len(s):
            if i < len(s)-1:
                c = s[i] + s[i+1]
            if c in romanLetters.keys():
                val += romanLetters[c]
                i+=2
            else:
                val += romanLetters[s[i]]
                i+=1
            c = ""
        return val

File: Top_Interview_150_List_With_solution(leetcode)/Array|String/L13-RomanToInteger.java
// https://leetcode.com/problems/roman-to-integer/
class Solution {
    public void fillMap(Map<String, Integer> m) {
        m.put("I", 1);
        m.put("V", 5);
        m.put("X", 10);
        m.put("L", 50);
        m.put("C", 100);
        m.put("D", 500);
        m.put("M", 1000);
        m.put("IV", 4);
        m.put("IX", 9);
        m.put("XL", 40);
        m.put("XC", 90);
        m.put("CD", 400);
        m.put("CM", 900);        
    }
    public int romanToInt(String s) {
        Map<String, Integer> romanToInteger = new HashMap<>();
        int val = 0, i = 0, length = s.length();
        fillMap(romanToInteger);
        while (i < length) {
            String c = "";
            if (i + 1 < length) {
                c = s.charAt(i) + "" + s.charAt(i + 1);
            }
            int t = romanToInteger.getOrDefault(c, 0);
            if (t > 0) {
                val += t;
                i += 2;
            } else {
                val += romanToInteger.getOrDefault(s.charAt(i) + "", 0);
                i += 1;
            }
        }
        return val;
    }
}
